egs/heroico/s5/run.sh

```diff
@@ -0,0 +1,295 @@
+#!/bin/bash
+# -*- tab-width: 2; indent-tabs-mode: nil; -*-
+
+. ./cmd.sh
+
+. ./path.sh
+stage=0
+
+. ./utils/parse_options.sh
+
+set -e
+set -o pipefail
+set -u
+
+# the location of the LDC corpus
+datadir=/mnt/corpora/LDC2006S37/data
+
+# location of subs text data
+subsdata="http://opus.lingfil.uu.se/download.php?f=OpenSubtitles2016/en-es.txt.zip"
+lexicon="http://www.openslr.org/resources/34/santiago.tar.gz"
+tmpdir=data/local/tmp
+
+if [ $stage -le 0 ]; then
+  # prepare the lists for acoustic model training and testing
+  mkdir -p $tmpdir/heroico
+  mkdir -p $tmpdir/usma
+
+  [ ! -d "$datadir" ] && \
+    echo "Data directory (LDC corpus release) does not exist" && \
+    exit 1
+  local/prepare_data.sh $datadir
+fi
+
+if [ $stage -le 1 ]; then
+  # prepare a dictionary
+  mkdir -p data/local/dict
+  mkdir -p data/local/tmp/dict
+
+  # download the dictionary from openslr
+  if [ ! -f data/local/tmp/dict/santiago.tar.gz ]; then
+    wget -O data/local/tmp/dict/santiago.tar.gz $lexicon
+  fi
+
+  (
+    #run in shell, so we don't have to remember the path
+    cd $tmpdir/dict
+    tar -xzf santiago.tar.gz
+  )
+
+  local/prepare_dict.sh
+
+  # prepare the lang directory
+  utils/prepare_lang.sh \
+    data/local/dict "<UNK>" \
+    data/local/lang data/lang   || exit 1;
+fi
+
+if [ $stage -le 2 ]; then
+    # use am training text to train lm
+  mkdir -p $tmpdir/heroico/lm
+
+  cut \
+      -d " " \
+      -f 2- \
+      data/train/text \
+      > \
+      $tmpdir/heroico/lm/train.txt
+
+  # get subs data for lm training
+  mkdir -p $tmpdir/subs/lm
+
+  # download  subs text data
+  if [ ! -f $tmpdir/subs/es.zip ]; then
+  wget \
+      -O $tmpdir/subs/es.zip \
+      $subsdata
+  fi
+
+  (
+    cd $tmpdir/subs
+    unzip es.zip
+    # delete parallel parts of the subs corpus
+    rm es.zip OpenSubtitles2016.en-es.en OpenSubtitles2016.en-es.ids
+  )
+
+  lm_training_data=$tmpdir/subs/OpenSubtitles2016.en-es.es
+fi
+
+#if [ $stage -le 3 ]; then
+  # get a sample of the subs corpus for lm training
+  #local/subs_prepare_data.pl
+
+  #rm $tmpdir/subs/OpenSubtitles2016.en-es.es
+#fi
+
+if [ $stage -le 4 ]; then
+  # build lm
+  local/prepare_lm.sh $lm_training_data
+
+  utils/format_lm.sh \
```

Comment:
Please don't put each argument on a separate line like this -- it makes the script hard to read without excessive scrolling.